### PR TITLE
feat: hex code color highlight

### PR DESCRIPTION
**Pull request by SyedMuzamilM**
If AI returns a list of hex code colors then you will see the color highlighted after the text so that you can easily review the color and select the color you want.

color-highlight

### CODE REVIEW

**Comment by saliksik** [comment]
if the HEX code is formatted this way, the color highlight does not work

Screenshot (306)

Screenshot (307)

**Comment by SyedMuzamilM** [comment]
Will check and fix.

**Comment by SyedMuzamilM** [comment]
I think this is a code block should this also have a highlight after it

File: components/Chat/ChatMessage.tsx [hunk 1]
```diff
@@ -212,6 +209,28 @@ export const ChatMessage: FC<Props> = memo(
                         </td>
                       );
                     },
+                    li({ children }) {
+                      const colorMatch =
```
Comment (SyedMuzamilM) [inline components/Chat/ChatMessage.tsx:213]: This is how I can modify the code to check if it contains a code element and then check for the hex code. Is it ok to add here then we will get the color highlight if AI returns the hex color in code element or ``

``` typescript
const regex = /#([A-Fa-f0-9]{6}|[A-Fa-f0-9]{3})/g;
  let colorMatch = regex
    .exec(
      String(children),
    );

  if (children?.length > 1) {
    Children.forEach(children, (child) => {
      if (isValidElement(child) && child.props.node.tagName === 'code') {
        console.log({ child });
        const match = regex.exec(child.props.children);
        if (match) {
          colorMatch = match;
        }
      }
    })
  }
```

**Comment by mckaywrigley** [comment]
I think this is a really cool idea, but I don't think we will support this by default.

I would save this code for when plugins come on Chatbot UI. You can build a hex color preview plugin that users can add to their ui :)

**Comment by SyedMuzamilM** [comment]
Awesome idea.

**Pull request closed without merging**